### DISCUSSION TITLE
Is it possible to access the actual GrpcServer Instance ?

### DISCUSSION THREAD

**Discussion by sdimitris** [deeplay-io/nice-grpc#107] [Q&A]
Hello there.
I was wondering if the `GrpcServer`( @grpc/grpc-js instance)  actual Instance can be accessed somehow.
Is there a way to expose this instance using  **nice-grpc** `Server<{}>` instance?

**Comment by aikoven** [comment]
The `grpc-js` is hidden as an implementation detail. What exactly do you want to achieve?

**Reply by sdimitris** [reply]
I want to add services in the server in two ways.
The first one is the classic one with grpc.addService
The second one with the nice-grpc **add** function.

**Reply by aikoven** [reply]
Could you please elaborate on why do you need this? Is that about some legacy services that you want to add do the server?

The `grpc-js` server is an implementation detail that we wouldn't like to expose, because it would create a tight coupling to `grpc-js` API. While it is hidden, we are free to replace the implementation in the future, possibly with our custom gRPC implementation.

We could expose the `grpc-js` server instance with no guarantees, in a field like `_unstable_grpc_js_server`, but I'd like to first be sure that there's no other option for you.

One of such options could be an HTTP2 proxy that forwards traffic to one of the two servers listening on different ports.
